### PR TITLE
Add typings

### PR DESCRIPTION
**Pull request by pellejacobs**
Adding typescript type declarations, closing #31 

Also added `jasmine-core` to dev dependencies as it seemed to be missing and added `.editorconfig`.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+root = true
+
+[*]
+indent_style = space
+indent_size = 4
+charset = utf-8
+trim_trailing_whitespace = true
```
Comment (prettymuchbryce) [inline .editorconfig]: Lets remove editorconfig for now.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+   * You can change the number of calculations done in a call by using
+   * easystar.setIteratonsPerCalculation().
+   */
+  calculate(): void
+}
+
+
+
+
+
```
Comment (prettymuchbryce) [inline index.d.ts]: Why so much extra space down here ?

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
+export class js {
+
+  /**
+  * Sets the collision grid that EasyStar uses.
+  * 
+  * @param {Array|Number} tiles An array of numbers that represent
```
Comment (prettymuchbryce) [inline index.d.ts]: Array.<number>

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  /**
+  * Sets the collision grid that EasyStar uses.
+  * 
+  * @param {Array|Number} tiles An array of numbers that represent 
+  * which tiles in your grid should be considered
+  * acceptable, or "walkable".
+  */
+  setAcceptableTiles(tiles: number[] | number): void
+  /**
+    * Enables sync mode for this EasyStar instance..
```
Comment (prettymuchbryce) [inline index.d.ts]: Can you fix the spacing on this comment block.

File: index.d.ts (hunk 2 above)
Comment (pellejacobs) [inline index.d.ts]: Typings need a LOT of space 😐

**Comment by prettymuchbryce** [comment]
I'm not an active TypeScript user. How would I test this out ?

I installed VS code and created this snippet on your branch, but it doesn't seem to be suggesting the methods to me:
```
/// <reference path="./easystarjs/index.d.ts" />
var EasyStar = require('easystar.js');
var easystar = new EasyStar.js();

easystar.
```

File: index.d.ts [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+  /**
+   * Disable diagonal pathfinding.
+   */
+  disableDiagonals(): void
+  
+  /**
+   * Sets the collision grid that EasyStar uses.
+   * 
+   * @param {Array} grid The collision grid that this EasyStar instance will read from.
```
Comment (prettymuchbryce) [inline index.d.ts]: Does TypeScript use these comments, or does it pull the comments from the `.js` itself ? I hate to have to duplicate documentation.

**Comment by pellejacobs** [comment]
I'm not familiar with jasmine testing. I can try to figure it out, but for sure, you'll have to pull in the typescript module. Here is a repo with typings I wrote for another library, it has tests included: https://github.com/pellejacobs/typed-redux-persist

File: index.d.ts (hunk 5 above)
Comment (pellejacobs) [inline index.d.ts]: So I now vscode's intellisense uses the documentation from the typings file. In general, it seems to be common practice to put all documentation in the typings file, take eg. redux's typings: https://github.com/reactjs/redux

Comment (prettymuchbryce) [inline index.d.ts]: Ok fair enough. 👍

Comment (pellejacobs) [inline index.d.ts]: So just leave it like this?

**Comment by prettymuchbryce** [comment]
I managed to get this working locally. Looks good! I will merge now.

**Pull request merged**